src/operations.cc

```diff
@@ -212,6 +212,25 @@ namespace sharp {
   }
 
   /*
+   * Convolution with a kernel.
+   */
+  VImage Convolve(VImage image, int width, int height, double scale, double offset, std::vector<double> kernel_v) {
+    VImage kernel = VImage::new_matrix(width, height);
```

Comment:
Might it be possible to use [new_from_memory](https://github.com/jcupitt/libvips/blob/master/cplusplus/include/vips/VImage8.h#L440) instead here? `NAN_METHOD(pipeline)` already contains a loop over the values of the kernel array - if it built a `char*` instead of a `std::vector` the loop and pointer logic would no longer be required here.